plugins/crypto/openssl/certificategroup.c

```diff
@@ -306,13 +307,17 @@ reloadCertificates(UA_CertificateGroup *certGroup) {
         return UA_STATUSCODE_BADOUTOFMEMORY;
     }
     for(size_t i = 0; i < context->trustList.trustedCrlsSize; i++) {
-        const unsigned char *pData = context->trustList.trustedCrls[i].data;
         X509_CRL * crl = NULL;
+        BIO * bio = NULL;
 
-        if(context->trustList.trustedCrls[i].length > 1 && pData[0] == 0x30 && pData[1] == 0x82) { // Magic number for DER encoded files
-            crl = d2i_X509_CRL (NULL, &pData, (long)context->trustList.trustedCrls[i].length);
-        } else {
-            BIO* bio = NULL;
+        /* Try to load DER encoded CRL */
+        bio = BIO_new_mem_buf((void *)context->trustList.trustedCrls[i].data,
+                              (int)context->trustList.trustedCrls[i].length);
+        crl = d2i_X509_CRL_bio(bio, NULL);
+        BIO_free(bio);
```

Comment:
We can simply reuse the bio if CRL == NULL.

Reply:
done